### ISSUE TITLE
v2: Beta 16 will auto-migrate certificate assets to a new folder path

### ISSUE THREAD

**Issue by mholt** [caddyserver/caddy#3124]
In beta 16 (starting with commit 4d18587192e4fffe5b34b714eaabcfc212914c1e with a couple of fixes added in 217419f6d97885daf0fb8832a407626e578523f0), Caddy will store certificates in a new path, and will attempt to automatically move them for you.

- **Old paths: `<caddy data dir>/acme/<ca>/sites/<domain>`**
- **New paths: `<caddy data dir>/certificates/<ca>/<domain>`**

Where `<ca>` is derived from the CA's directory endpoint. For CAs containing `api.letsencrypt.org`, the new path appends `-directory` to the end, for example: `acme-v02.api.letsencrypt.org-directory`.

This move is necessary to better support non-ACME certificate sources. (We have done similar moves in the past and it's always worked out alright.)

The structure of the `.json` resource file associated with each certificate has also changed slightly, and it will be re-encoded for you. (This file is non-critical.)

The parent data directory has _not_ changed, so this is a move of subfolders within a singular parent directory.

## If you have configured custom storage

If you are not using the file system for storage, you will need to move those certificates yourself, or just let Caddy get new ones. Up to a moderate number of certificates (~50) this should be totally fine.

If you are using the file system but have configured your own path for storage, Caddy will operate the migration within that custom path.

## If there are errors

Errors during the auto-migration are non-critical. The worst thing that will probably happen as a resul of errors is that Caddy will attempt to get new certificates instead of using existing ones. Unless you have hundreds of certificates, this shouldn't be a problem.

## tl;dr

This post is mostly informational, not a call to action. Most users will not have to do anything. You'll see in the logs if a migration occurs. If there are errors, they will be printed to the log. Errors are non-critical. If you have problems, feel free to [post on the forums](https://caddy.community) or reply to this thread.

Closing this issue but feel free to comment if necessary.
